### PR TITLE
Add Stack Overflow tag component

### LINKED ISSUE #4720: "Ask on Stack Overflow" include
**Is your feature request related to a problem? Please describe.**
At times, the articles we publish are related to technologies that have "official" tags on Stack Overflow. (E.g. [`service-worker`](https://stackoverflow.com/questions/tagged/service-worker).)

**Describe the solution you'd like**
 It would be nice if there were some sort of "include" snippet that could be added to a post and take in the Stack Overflow tag as a parameter, and produce HTML along the lines of the following, assuming `{{tag}}` is a placeholder for the `tag` property:

```
<p>
  Have a question about using this feature?
  You can get help by
  <a href="https://stackoverflow.com/questions/ask?tags={{tag}}">asking a question on Stack Overflow</a>,
  or <a href="https://stackoverflow.com/search?q=%{{tag}}%5D+is%3Aquestion">browse a list of questions</a>
  asked by other developers.
</p>
```

(Obviously feel free to tweak this based on style, etc.)

**Describe alternatives you've considered**
We could hardcode this text into each post, but that would not be as easy to use and it would make it hard to tweak the text in a standard manner.

**Additional context**
I think this is conceptually similar to the origin trial and Glitch includes that are already established on the site.

CC: @petele

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Sometimes, a post is related to technologies that have "official" tags on Stack Overflow (e.g. `service-worker`).

This PR is adding a brand new Stack Overflow tag component that when used will show an "Ask on Stack Overflow" note the end of the post.

Live preview: 
- https://deploy-preview-4721--web-dev-staging.netlify.app/handbook/yaml-front-matter/#stack_overflow_tag
- https://deploy-preview-4721--web-dev-staging.netlify.app/nfc/#acknowledgements

Fixes #4720

### CODE REVIEW

File: src/site/_includes/components/StackOverflow.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const {html} = require('common-tags');
+
+module.exports = (tag) => {
+  return html`
+    <div class="w-aside w-aside--note">
+      Have a question about using this feature? You can get help by
+      <a href="https://stackoverflow.com/questions/ask?tags=${tag}"
+        >asking a question on Stack Overflow</a
+      >, or
+      <a href="https://stackoverflow.com/search?q=%${tag}%5D+is%3Aquestion"
```
Comment (tomayac) [inline src/site/_includes/components/StackOverflow.js:26]:
```suggestion
      <a href="https://stackoverflow.com/search?q=%5B${tag}%5D+is%3Aquestion"
```

Fix the percent encoding.

File: src/site/_drafts/_template-post/index.md [hunk 2]
```diff
@@ -63,6 +63,10 @@ tags:
   - blog # blog is a required tag for the article to show up in the blog.
   - accessibility
   - ux
+
+# You can provide an optional Stack Overflow tag to be used at the end of the
+# post in a "Ask on Stack Overflow" note.
```
Comment (tomayac) [inline src/site/_drafts/_template-post/index.md:68]:
```suggestion
# post in an "Ask on Stack Overflow" note.
```

I know, NittyMcNitFace

File: src/site/content/en/handbook/yaml-front-matter/index.md [hunk 3]
```diff
@@ -216,6 +216,26 @@ example.
...
+
+```yaml
+---
+…
+stack_overflow_tag: service-worker
+…
+---
+```
+
+Sometimes, a post is related to technologies that have "official" tags on Stack Overflow
```
Comment (tomayac) [inline src/site/content/en/handbook/yaml-front-matter/index.md:229]:
```suggestion
Sometimes, a post is related to technologies that have
["official" tags on Stack Overflow](https://stackoverflow.com/tags)
```

File: src/site/content/en/handbook/yaml-front-matter/index.md [hunk 4]
```diff
@@ -216,6 +216,26 @@ example.
...
+(e.g. `service-worker`). When used, an "Ask on Stack Overflow" note will be shown at the end
```
Comment (tomayac) [inline src/site/content/en/handbook/yaml-front-matter/index.md:230]:
```suggestion
(e.g. [`service-worker`](https://stackoverflow.com/questions/tagged/service-worker)).
When used, an "Ask on Stack Overflow" note will be shown at the end
```

File: src/site/_drafts/_template-post/index.md [hunk 5]
```diff
@@ -63,6 +63,10 @@ tags:
...
+# stack_overflow_tag: service-worker
```
Comment (tomayac) [inline src/site/_drafts/_template-post/index.md:69]:
```suggestion
# Find the official list of tags at https://stackoverflow.com/tags.
# stack_overflow_tag: service-worker
```

File: src/site/_includes/components/StackOverflow.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        >browse a list of questions</a
```
Comment (tomayac) [inline src/site/_includes/components/StackOverflow.js:27]:
```suggestion
        >browsing a list of questions</a
```

**Review by tomayac — CHANGES_REQUESTED** [review]
Just some additional improvements.

File: src/site/_includes/components/StackOverflow.js (hunk 1 above)
Comment (beaufortfrancois) [reply src/site/_includes/components/StackOverflow.js]: Thanks!

File: src/site/_includes/components/StackOverflow.js (hunk 6 above)
Comment (beaufortfrancois) [reply src/site/_includes/components/StackOverflow.js]: Yep!

File: src/site/content/en/handbook/yaml-front-matter/index.md (hunk 3 above)
Comment (beaufortfrancois) [reply src/site/content/en/handbook/yaml-front-matter/index.md]: Done

File: src/site/content/en/handbook/yaml-front-matter/index.md (hunk 4 above)
Comment (beaufortfrancois) [reply src/site/content/en/handbook/yaml-front-matter/index.md]: Done.

**Review by samthor — COMMENTED** [review]
I'm quite supportive of this.

- should the image in the handbook be in the storage bucket
- is there a chance we might want to reference multiple tags, or is this YAGNI?

File: src/site/content/en/handbook/yaml-front-matter/index.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ layout: handbook
 title: YAML front matter
 subhead: |
   Reference documentation for web.dev's YAML front matter properties.
-date: 2020-08-06
+date: 2021-02-23
```
Comment (tomayac) [inline src/site/content/en/handbook/yaml-front-matter/index.md:6]:
```suggestion
date: 2021-03-03
```

**Comment by tomayac** [comment]
> I'm quite supportive of this.

Yay!

> * should the image in the handbook be in the storage bucket

Done in https://github.com/GoogleChrome/web.dev/pull/4721/commits/d58a8053b2638fa5a00d84fbc3fe265a7066a45f.

> * is there a chance we might want to reference multiple tags, or is this YAGNI?

I think this is YAGNI. @beaufortfrancois?

**Comment by beaufortfrancois** [comment]
> I think this is YAGNI. @beaufortfrancois?

+1

**Review by robdodson — APPROVED** [review]

**Comment by robdodson** [comment]
I think this is good to land once the tests pass. @beaufortfrancois feel free to hit merge when you're ready.

**Comment by beaufortfrancois** [comment]
Percy jobs still fail sadly. And I'm not sure if my change is related to this failure sadly ;(
```
> web.dev@3.0.0 snapshots /home/runner/work/web.dev/web.dev
> percy exec --config ./tools/percy/.percy.conf.yml -- node tools/percy/snapshots.js

 ›   Warning: Skipping visual tests. PERCY_TOKEN was not provided.
Error: net::ERR_CONNECTION_REFUSED at http://localhost:8080/
    at navigate (/home/runner/work/web.dev/web.dev/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:113:23)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async FrameManager.navigateFrame (/home/runner/work/web.dev/web.dev/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:88:21)
    at async Frame.goto (/home/runner/work/web.dev/web.dev/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:405:16)
    at async Page.goto (/home/runner/work/web.dev/web.dev/node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js:826:16)
    at async PercyScript.run.args (/home/runner/work/web.dev/web.dev/tools/percy/snapshots.js:97:7)
    at async Object.run (/home/runner/work/web.dev/web.dev/node_modules/@percy/script/index.js:25:7)
```

@robdodson May you be able to help?

**Comment by beaufortfrancois** [comment]
I've recreated PR at https://github.com/GoogleChrome/web.dev/pull/4832 to make Percy jobs pass.

**Pull request merged**